recipes/jsoncpp/all/conanfile.py

```diff
@@ -0,0 +1,67 @@
+from conans import ConanFile, CMake, tools
+import os, shutil
+
+
+class JsoncppConan(ConanFile):
+    name = "jsoncpp"
+    license = "MIT"
+    homepage = "https://github.com/open-source-parsers/jsoncpp"
+    url = "https://github.com/conan-io/conan-center-index"
+    topics = ("conan", "json", "parser", "config")
+    description = "A C++ library for interacting with JSON."
+    settings = "os", "compiler", "arch", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+    exports_sources = "CMakeLists.txt",
+    generators  = "cmake",
+
+    _source_subfolder = "source_subfolder"
+
+    def configure(self):
+        if self.settings.os == "Windows" or self.options.shared:
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "jsoncpp-{}".format(self.version)
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _patch_sources(self):
+        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
+                              "${jsoncpp_SOURCE_DIR}",
+                              "${JSONCPP_SOURCE_DIR}")
+        if self.settings.compiler == "Visual Studio" and self.settings.compiler.version == "11":
+            tools.replace_in_file(os.path.join(self._source_subfolder, "include", "json", "value.h"),
+                                  "explicit operator bool()",
+                                  "operator bool()")
+
+        if self.settings.os != "Windows" and not self.options.shared and not self.options.fPIC:
+            tools.replace_in_file(os.path.join(self._source_subfolder, "src", "lib_json", "CMakeLists.txt"),
+                                  "set_target_properties( jsoncpp_lib PROPERTIES POSITION_INDEPENDENT_CODE ON)",
+                                  "set_target_properties( jsoncpp_lib PROPERTIES POSITION_INDEPENDENT_CODE OFF)")
+        tools.replace_in_file(os.path.join(self._source_subfolder, "src", "lib_json", "CMakeLists.txt"),
+                              "$<BUILD_INTERFACE:${PROJECT_BINARY_DIR}/include/json>",
+                              "")
+
+    def _get_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["JSONCPP_WITH_TESTS"] = False
+        cmake.definitions["JSONCPP_WITH_CMAKE_PACKAGE"] = False
+        cmake.definitions["JSONCPP_WITH_STRICT_ISO"] = False
+        cmake.definitions["JSONCPP_WITH_PKGCONFIG_SUPPORT"] = False
+        return cmake
+
+    def build(self):
+        self._patch_sources()
+        cmake = self._get_cmake()
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
+
+        cmake = self._get_cmake()
+        cmake.install()
+    def package_info(self):
+        self.cpp_info.name = "JsonCPp"
```

Comment:
Not sure, but this name looks wrong
```suggestion
        self.cpp_info.name = "JsonCpp"
```

Reply:
This looks very wrong indeed :smile: